src/main/java/sg/ncl/MainController.java

```diff
@@ -2003,16 +2017,50 @@ public String joinTeamAppAwaitingApproval(Model model) {
     }
     
     //--------------------------Get List of scenarios filenames--------------------------
-    private List<String> getScenarioFileNameList() {
-		List<String> scenarioFileNameList = new ArrayList<String>();
-		File[] files = new File(SCENARIOS_DIR_PATH).listFiles();
-		for (File file : files) {
-			if (file.isFile()) {
-				scenarioFileNameList.add(file.getName());
-			}
-		}
+    private List<String> getScenarioFileNameList() throws WebServiceRuntimeException {
+        logger.info("Retrieving scenario file names");
+//        List<String> scenarioFileNameList = null;
+//        try {
+//            scenarioFileNameList = IOUtils.readLines(getClass().getClassLoader().getResourceAsStream("scenarios"), StandardCharsets.UTF_8);
+//        } catch (IOException e) {
+//            throw new WebServiceRuntimeException(e.getMessage());
+//        }
+//        File folder = null;
+//        try {
+//            folder = new ClassPathResource("scenarios").getFile();
+//        } catch (IOException e) {
+//            throw new WebServiceRuntimeException(e.getMessage());
+//        }
+//        List<String> scenarioFileNameList = new ArrayList<>();
+//		File[] files = folder.listFiles();
+//		for (File file : files) {
+//			if (file.isFile()) {
+//				scenarioFileNameList.add(file.getName());
+//			}
+//		}
+        // FIXME: hardcode list of filenames for now
+        List<String> scenarioFileNameList = new ArrayList<>();
+        scenarioFileNameList.add("basic.ns");
+        scenarioFileNameList.add("basic2.ns");
+        logger.info("Scenario file list: {}", scenarioFileNameList);
 		return scenarioFileNameList;
     }
+
+    private String getScenarioContentsFromFile(String scenarioFileName) throws WebServiceRuntimeException {
+        try {
+            logger.info("Retrieving scenario files {}", getClass().getClassLoader().getResourceAsStream("scenarios/" + scenarioFileName));
+            List<String> lines = IOUtils.readLines(getClass().getClassLoader().getResourceAsStream("scenarios/" + scenarioFileName), StandardCharsets.UTF_8);
+            StringBuilder sb = new StringBuilder();
+            for (String line : lines) {
+                sb.append(line);
+                sb.append(System.getProperty("line.separator"));
+            }
+            logger.info("Experiment ns file contents: {}", sb);
+            return sb.toString();
+        } catch (IOException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)